### ISSUE TITLE
What are offsets relative to?

### ISSUE THREAD

**Issue by eqrion** [WebAssembly/branch-hinting#25]
The overview states:

```
Each branch hint structure consists of

    the |U32| byte offset of the hinted instruction from the beginning of the function body,
```

What is the beginning of the function body? Is that before the vector of locals or after the vector of locals?

**Comment by yuri91** [comment]
just after the size field, corresponding with the beginning of the vector of locals.

**Comment by eqrion** [comment]
Would you be able to update the overview or spec for this? We accidentally implemented this the wrong way in SM.

**Comment by yuri91** [comment]
Yes. Sorry for the confusion, it is indeed not very clear. It also used to be the other way around earlier in the design.

**Comment by yuri91** [comment]
Looking into how to improve the wording, I realized that the spec already has a concept of the function "body":

> the actual function code, which in turn consists of
>
> - the declaration of locals,
> 
> - the function **body** as an *expression*


which is at odds with the usage in this proposal.

I will change the overview like this:

> * the |U32| byte offset of the hinted instruction from the beginning of the function body 

->

> * the |U32| byte offset of the hinted instruction, relative to the beginning of the function locals declaration

and similarly the spec text.

Thanks for bringing this up!

**Comment by rossberg** [comment]
Why not just replace "body" with "code" to match the terminology defined in the spec?

**Comment by yuri91** [comment]
@rossberg for the overview,  i think that the term *code* would not be clear enough on its own, without the surrounding context (especially because *code* is both the name of the production rule and the name of the field). But I can expand the description if this is not enough.

For the spec, I opted to use the |Bfunc| term, because it becomes a link to the actual definition in the main spec. The "code" field is defined as |Bfunc|, so it should be more or less in line with your suggestion,  i think?

**Comment by rossberg** [comment]
Makes sense, thanks.
